OMX-27-firmware/OMX-27-firmware.ino

```diff
@@ -1070,113 +1185,116 @@ void loop() {
 							}
 							rotationAmt = constrain(rotationAmt, (PatternLength(playingPattern)-1)*-1, PatternLength(playingPattern)-1);
 						}
-						if (ppmode == 3) { 					// SET PATTERN CHANNEL
+						if (ppparam == 4) { 					// SET PATTERN CHANNEL
 							patternSettings[playingPattern].channel = constrain(patternSettings[playingPattern].channel + amt, 0, 15);
 						}
-
-						if (ppmode3 == 0) { 					// SET CLOCK-DIV-MULT
-							patternSettings[playingPattern].clockDivMultP = constrain(patternSettings[playingPattern].clockDivMultP + amt, 0, NUM_MULTDIVS-1); // set clock div/mult
-						}
-						if (ppmode3 == 1) { 					// SET MIDI SOLO
-							patternSettings[playingPattern].solo = constrain(patternSettings[playingPattern].solo + amt, 0, 1);
-						}
-
-						// PATTERN PARAMS PAGE 2
-							//TODO: convert to case statement ??
-						if (ppmode2 == 0) { 					// SET AUTO START STEP
+						// PAGE THREE
+						if (ppparam == 6) { 					// SET AUTO START STEP
 							patternSettings[playingPattern].startstep = constrain(patternSettings[playingPattern].startstep + amt, 0, patternSettings[playingPattern].len);
 							//patternSettings[playingPattern].startstep--;
 						}
-						if (ppmode2 == 1) { 					// SET AUTO RESET STEP
+						if (ppparam == 7) { 					// SET AUTO RESET STEP
 							int tempresetstep = patternSettings[playingPattern].autoresetstep + amt;
 							patternSettings[playingPattern].autoresetstep = constrain(tempresetstep, 0, patternSettings[playingPattern].len+1);
 						}
-						if (ppmode2 == 2) { 					// SET AUTO RESET FREQUENCY
+						if (ppparam == 8) { 					// SET AUTO RESET FREQUENCY
 							patternSettings[playingPattern].autoresetfreq = constrain(patternSettings[playingPattern].autoresetfreq + amt, 0, 15); // max every 16 times
 						}
-						if (ppmode2 == 3) { 					// SET AUTO RESET PROB
+						if (ppparam == 9) { 					// SET AUTO RESET PROB
 							patternSettings[playingPattern].autoresetprob = constrain(patternSettings[playingPattern].autoresetprob + amt, 0, 100); // never, 100% - 33%
 						}
+						// PAGE THREE
+						if (ppparam == 11) { 					// SET CLOCK-DIV-MULT
+							patternSettings[playingPattern].clockDivMultP = constrain(patternSettings[playingPattern].clockDivMultP + amt, 0, NUM_MULTDIVS-1); // set clock div/mult
+						}
+						if (ppparam == 12) { 					// SET MIDI SOLO
+							patternSettings[playingPattern].solo = constrain(patternSettings[playingPattern].solo + amt, 0, 1);
+						}
 
-					} else if (stepRecord && !enc_edit){	// STEP RECORD MODE
-
-						if (srmode == 4 && srmode2 == 4) { 	// CHANGE PAGE
+					// STEP RECORD MODE
+					} else if (stepRecord && !enc_edit){
+						// CHANGE PAGE
+						if (srparam == 0 || srparam == 5) { 	
 							srpage = constrain(srpage + amt, 0, 1);		// HARDCODED - FIX WITH SIZE OF PAGES?
 						}
-						if (srmode == 3) {
-//							playingPattern = constrain(playingPattern + amt, 0, 7);
+						srparam = srpage * NUM_DISP_PARAMS;
```

Comment:
Move ```srparam = srpage * NUM_DISP_PARAMS;``` into the if condition above.

I also provided a couple similar comments above.